astropy/table/operations.py

```diff
@@ -339,45 +339,125 @@ def unique(input_table, keys=None, silent=False):
         Default is to use all columns.
+    keep : one of 'first', 'last' or 'none'
+        Whether to keep the first or last row for each set of
+        duplicates. If 'none', all rows that are duplicate are
+        removed, leaving only rows that are already unique in
+        the input.
+        Default is 'first'.
     silent : boolean
-        If `True` masked value column(s) are silently removed from
-        ``keys``. If `False` an exception is raised when ``keys`` contains
-        masked value column(s).
+        If `True`, masked value column(s) are silently removed from
+        ``keys``. If `False`, an exception is raised when ``keys``
+        contains masked value column(s).
         Default is `False`.
 
     Returns
     -------
     unique_table : `~astropy.table.Table` object
-        Table containing only the unique rays of ``input_table``.
+        New table containing only the unique rows of ``input_table``.
+
+    Examples
+    --------
+    >>> from astropy.table import Table
+    >>> import numpy as np
+    >>> table = Table(data=[[1,2,3,2,3,3],
+    ... [2,3,4,5,4,6],
+    ... [3,4,5,6,7,8]],
+    ... names=['col1', 'col2', 'col3'],
+    ... dtype=[np.int32, np.int32, np.int32])
+    >>> table
+    <Table length=6>
+     col1  col2  col3
+    int32 int32 int32
+    ----- ----- -----
+        1     2     3
+        2     3     4
+        3     4     5
+        2     5     6
+        3     4     7
+        3     6     8
+    >>> unique(table, keys='col1')
+    <Table length=3>
+     col1  col2  col3
+    int32 int32 int32
+    ----- ----- -----
+        1     2     3
+        2     3     4
+        3     4     5
+    >>> unique(table, keys=['col1'], keep='last')
+    <Table length=3>
+     col1  col2  col3
+    int32 int32 int32
+    ----- ----- -----
+        1     2     3
+        2     5     6
+        3     6     8
+    >>> unique(table, keys=['col1', 'col2'])
+    <Table length=5>
+     col1  col2  col3
+    int32 int32 int32
+    ----- ----- -----
+        1     2     3
+        2     3     4
+        2     5     6
+        3     4     5
+        3     6     8
+    >>> unique(table, keys=['col1', 'col2'], keep='none')
+    <Table length=4>
+     col1  col2  col3
+    int32 int32 int32
+    ----- ----- -----
+        1     2     3
+        2     3     4
+        2     5     6
+        3     6     8
+    >>> unique(table, keys=['col1'], keep='none')
+    <Table length=1>
+     col1  col2  col3
+    int32 int32 int32
+    ----- ----- -----
+        1     2     3
 
     """
 
+    if keep not in ('first', 'last', 'none'):
+        raise ValueError("'keep' should be one of 'first', 'last', 'none'")
+
     if keys is None:
         keys = input_table.colnames
 
     if input_table.masked:
         if isinstance(keys, six.string_types):
-            keys = [keys, ]
-        for i, key in enumerate(keys):
+            keys = [keys]
+        nkeys = 0
+        for i, key in enumerate(keys[:]):
```

Comment:
Good you noticed the latent bug in this loop to drop keys for masked columns (and thus changed `keys` to `keys[:]`).  But it still won't work if more than one key column gets deleted since the index `i` is out of sync with the original `keys`.  I leave it to you to figure out the right way and add a test.